test/tests-extension/util/resources.go

```diff
@@ -0,0 +1,91 @@
+package util
+
+import (
+	"context"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+// CreateTestNamespace creates a test namespace
+func CreateTestNamespace(client kubernetes.Interface, namespaceName string) (*corev1.Namespace, func(), error) {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}
+	ns, err := client.CoreV1().Namespaces().Create(context.TODO(), ns, metav1.CreateOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+	cleanup := func() {
+		// best-effort delete + wait for termination
+		_ = client.CoreV1().Namespaces().Delete(context.TODO(), ns.Name, metav1.DeleteOptions{})
+		// Block until the namespace is fully gone to avoid name collisions
+		// (ignore errors after timeout to avoid hanging teardown).
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+		defer cancel()
+		wait.PollImmediateUntil(2*time.Second, func() (bool, error) {
+			_, err := client.CoreV1().Namespaces().Get(context.TODO(), ns.Name, metav1.GetOptions{})
+			if errors.IsNotFound(err) {
+				return true, nil
+			}
+			return false, nil
+		}, ctx.Done())
+	}
+	return ns, cleanup, err
```

Comment:
You’re returning err even though you already checked it before defining cleanup.
At this point, err will always be nil.

Return explicitly nil for the error: